catalog/spatial/registry/registry-publication-action-provider/src/main/java/org/codice/ddf/registry/publication/action/provider/RegistryPublicationActionProvider.java

```diff
@@ -0,0 +1,297 @@
+                () -> federationAdminService.getRegistryMetacards();
+        List<Metacard> metacards = subject.doAs(subject, federationAdminServiceAction);
+        // To here
+
+        for (Metacard metacard : metacards) {
+            Attribute locations =
+                    metacard.getAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS);
+            Attribute registryIdAttribute =
+                    metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID);
+            if (registryIdAttribute == null || StringUtils.isBlank(registryIdAttribute.getValue()
+                    .toString())) {
+                LOGGER.warn("Warning metacard (id: {}} did not contain a registry id.",
+                        metacard.getId());
+                continue;
+            }
+            String registryId = registryIdAttribute.getValue()
+                    .toString();
+            if (locations != null) {
+                publications.put(registryId,
+                        locations.getValues()
+                                .stream()
+                                .map(Object::toString)
+                                .collect(Collectors.toList()));
+            } else {
+                publications.put(registryId, Collections.emptyList());
+            }
+        }
+    }
+
+    private Action getAction(String regId, String destinationId, boolean publish) {
+
+        URL url;
+        String title = publish ? PUBLISH_TITLE + destinationId : UNPUBLISH_TITLE + destinationId;
+        String description = publish ?
+                PUBLISH_DESCRIPTION + destinationId :
+                UNPUBLISH_DESCRIPTION + destinationId;
+        String operation = publish ? PUBLISH_OPERATION : UNPUBLISH_OPERATION;
+        String httpOp = publish ? HTTP_POST : HTTP_DELETE;
+        try {
+
+            String path = String.format("%s/%s/%s/%s",
+                    REGISTRY_PATH,
+                    regId,
+                    PUBLICATION_PATH,
+                    URLEncoder.encode(destinationId, "UTF-8"));
+            URI uri = new URI(SystemBaseUrl.constructUrl(path, true));
+            url = uri.toURL();
+
+        } catch (MalformedURLException | URISyntaxException | UnsupportedEncodingException e) {
+            LOGGER.error("Malformed URL exception", e);
+            return null;
+        }
+        String id = String.format("%s.%s.%s", getId(), operation, httpOp);
+        return new ActionImpl(id, title, description, url);
+    }
+
+    private <T> String getRegistryId(T subject) {
+        if (subject instanceof Metacard) {
+            Metacard metacard = (Metacard) subject;
+            if (metacard.getTags()
+                    .contains(RegistryConstants.REGISTRY_TAG)) {
+                Attribute registryIdAttribute =
+                        metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID);
+                if (registryIdAttribute != null) {
+                    return registryIdAttribute.getValue()
```

Comment:
Is it possible for a metacard to have a registry ID attribute with a null value?